server/src/main/java/org/opensearch/gateway/GatewayMetaState.java

```diff
@@ -258,6 +254,49 @@ public void start(
         }
     }
 
+    private ClusterState restoreClusterStateWithRetries(
+        RemoteStoreRestoreService remoteStoreRestoreService,
+        ClusterState clusterState,
+        String lastKnownClusterUUID
+    ) {
+        int maxAttempts = 5;
+        int delayInMills = 100;
+        for (int attempt = 1; attempt <= maxAttempts; attempt++) {
+            try {
+                return restoreClusterState(remoteStoreRestoreService, clusterState, lastKnownClusterUUID);
```

Comment:
add a info log for every attempt

Reply:
Maybe reuse some retryable entity already present in cluster manager

Reply:
@Bukhtawar I do not see a suitable retryable entity already present. Let me know if you have any reference.

Reply:
How about extending this Action https://github.com/opensearch-project/OpenSearch/blob/09bacee5fc85676e97bee6b4ad87dec35c6aa8cc/server/src/main/java/org/opensearch/action/support/RetryableAction.java#L59-L81

Reply:
@Bukhtawar This RetryableAction is mostly used for transport action to provide support for retries. It needs the threadpool as well to to execute the action asynchronously. Also we need to provide a total timeout during which the retries to happen.
We will need provide total timeout and add plumbing for providing threadpool to use this. But the total timeout is difficult to predict as it depends on the total number of files to download due to which the download time will vary.
Also, we need to way to wait for the action to complete as the download of remote state is pre-requisite to move forward.
I think we should keep the logic simple as it is now.